Lib/typing.py

```diff
@@ -2132,6 +2132,9 @@ def __getattr__(self, attr):
             return 'Annotated'
         return super().__getattr__(attr)
 
+    def __mro_entries__(self, bases):
+        return (self.__origin__,)
```

Comment:
What happens without this? It seems kind of random to me (but I didn't try to run the tests).

Reply:
@gvanrossum First about `__module__`: it is not truly covered by test: https://github.com/python/cpython/blob/c7f2686d41a9ef654797ecf2bb56e9876d57eee9/Lib/test/test_typing.py#L7133

Here `typing.Any.__module__` is `typing`, so passed, but errors in `int`. It is directly caused by:

https://github.com/python/cpython/blob/c7f2686d41a9ef654797ecf2bb56e9876d57eee9/Lib/typing.py#L1357-L1358

The idea of this PR is to add `_name` attribute to `_AnnotatedAlias` instance, since `_AnnotatedAlias.__origin__` is just the first argument and nowhere to determind where `Annotated` annotation from.

But with `_name` attribute, `_GenericAlias` will think the `__origin__` is generic ABCs, `Generic` will finally add to mros.

https://github.com/python/cpython/blob/c7f2686d41a9ef654797ecf2bb56e9876d57eee9/Lib/typing.py#L1270-L1280

So `__mro_entries__` overridden is required.